src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -1497,6 +1497,23 @@ private void PrepareContainers()
 
                 foreach (var mount in containerMounts)
                 {
+                    if (mount.Type == ContainerMountType.BindMount &&
+                        mount.UnixFileMode is not null &&
```

Comment:
Evaluate the `mount` with pattern matching.

```suggestion
                    if (mount is { Type: ContainerMountType.BindMount, UnixFileMode: not null } &&
```